### ISSUE TITLE
Layout UI : Allow adding visibility conditions for Node:NID and Node:Type for paths other than node/%

### ISSUE THREAD

**Issue by Graham-72** [backdrop/backdrop-issues#1997]
I am importing a D7 site into Backdrop. The existing blog posts are aliased as node/ and not blog/ and I want to be able to reference them in a visibility condition in my layout, but there doesn't seem to be an option to do that.
![visibilitycondition](https://cloud.githubusercontent.com/assets/10559274/16538503/06f4f636-401e-11e6-8162-ab1978da5717.jpg)
So, I guess I will have to bulk delete the existing aliases and regenerate using the new pattern. But then, I think, I will lose the existing links known to search engines, won't I?

**Comment by klonos** [comment]
So do I get this right that you are trying to use `blog/%` as path instead of `node/%`?

**Comment by Graham-72** [comment]
To clarify, from the Drupal site I have imported the blogs that do not have node or blog in their URLs because they have been given the basic alias pattern [node:title] so a selector node/% does not distinguish them from all other nodes, unless we can have a visibility conditions like Node: Blog post, Node: Basic page etc.

**Comment by klonos** [comment]
Yeah, I know that. "Node:NID" and "Node:Type" are not made available unless the path is set to `node/%`. I know not of any other way.

**Comment by docwilmot** [comment]
@Graham-72 you could make a module thats adds this condition. It would be a few minutes job for you I'm sure. Look at my block_scheduler module as an example. https://github.com/backdrop-contrib/block_scheduler. The `.._layout_access` in that module is much more complex even than you'd need. See core front page access for a simpler example. https://github.com/backdrop/backdrop/blob/1.x/core/modules/layout/plugins/access/front_layout_access.inc

**Comment by serundeputy** [comment]
@Graham-72 if you go the route of generating new aliases you could generate redirects for the eld paths to the new.

here is how I do it: https://github.com/serundeputy/migrate_redirects

It is not user friendly, but gets the job done.  It is not a module you install, but temporarily put it in your `docroot/modules` folder just to use it.

* create a `redirects.csv` file one entry per line of:

```
eld_path, new_path
```

make sure to put the `redirects.csv` file into the `data` folder of the `migrate_redirects` code folder.
then run with the command `drush scr create_redirects.php` (cd into the migrate_redirects folder).

looks like you have a couple options here ;)

**Comment by klonos** [comment]
This is less than idea and we need to figure a better, UI way for this. I came across this issue in the past while testing - not on an actual use case or website. @Graham-72 though has a real use case (something that I was afraid would surface some day) and this means that others are likely to hit this wall. I don't think it's an extreme case.

I understand that contexts are a hard concept for people to grasp and the way we have them "automagically" configured depending on the path entered is really cool, but it obviously has its limitations.

**Comment by Graham-72** [comment]
Thanks everyone for replies and suggestions.

> for paths other than node/%

@klonos at first I was not understanding this. In this case I am working with the Default Layout and the UI has not mentioned node/%. But now the penny has dropped and looking at the custom layout I have used for the Home Page of the site I see that there I use node/% in the settings. So now I am thinking perhaps I could use a custom layout for the blog pages and thereby get access to the Node:Type option. But I admit that I have never yet fully understood the significance of the Path and Contexts settings in the UI, so maybe this is yet another opportunity to learn something.

**Comment by klonos** [comment]
> But I admit that I have never yet fully understood the significance of the Path and Contexts settings in the UI, so maybe this is yet another opportunity to learn something.

...on the same boat 😄

**Comment by klonos** [comment]
...I had many cases of customers where I had to take their desktop computers off-site for repair and they always asked: "Don't you need the keyboard/mouse?" I said "Nope, got plenty of those back in the lab.". They continued: "What about the monitor?". "Got plenty of those too". "But what about the icons I have on the desktop?" ...they thought that the icons came with the monitor. So basically, they were using their computer, but how it worked "behind the screen" was a total mystery to them.

That's how I feel about setting up the path and contexts in the Layout UI. It just magically works for me 😆

**Comment by Graham-72** [comment]
Solved it - quite simply!

I have defined a new layout 'Blog post' using the same layout template
![listlayouts](https://cloud.githubusercontent.com/assets/10559274/16541913/dd01b7ca-408a-11e6-951c-d04ad4acf2a3.png)

Then set its visibility condition to 'Node is Blog post'
![bloglayoutsettings](https://cloud.githubusercontent.com/assets/10559274/16541916/46a0e1f6-408b-11e6-893e-80d8c56938f9.png)

Then modified this layout 'Blog post' to include the blocks I need, but without a visibilty condition since that is controlled for the layout as a whole. Though the options now include Node:Type
![bloglayoutsettings2](https://cloud.githubusercontent.com/assets/10559274/16541965/046fd11e-408d-11e6-9e74-96562e48bf8a.jpg)


This is very useful and does all I need. But I am left wondering why the default layout is different.

**Comment by klonos** [comment]
> But I am left wondering why the default layout is different.

I think that @quicksketch explained this in a way that made perfect sense last time it happened to me and was wondering about the same thing. Unfortunately, it seems that I forgot what the explanation is.

**Comment by docwilmot** [comment]
Let's see if my new docs will help: https://backdropcms.org/user-guide/layouts

**Comment by Graham-72** [comment]
> Unfortunately, it seems that I forgot what the explanation is

There is too much to remember, and this reminds me to check out the emerging user guide.

**Comment by stpaultim** [comment]
@Graham-72 If I understand this issue correctly, the feature asked for in core is already there or possible and this issue can be closed.
